replication/binlogsyncer.go

```diff
@@ -178,6 +178,16 @@ func (b *BinlogSyncer) close() {
 		b.c.SetReadDeadline(time.Now().Add(100 * time.Millisecond))
 	}
 
+	// kill last connection id
+	if b.lastConnectionID > 0 {
```

Comment:
why use a new connection, cause it is easier to succeed?

Reply:
Without a new connection, calling `canal.Close()` leaves a binlog connection hanging. See #361.